monitors/BalanceMonitor.js

```diff
@@ -1,36 +1,63 @@
+const { createObjectFromDefaultProps } = require("../common/ObjectUtils");
 
 class BalanceMonitor {
   /**
    * @param {Object} logger Winston module used to send logs.
    * @param {Object} tokenBalanceClient Client used to query token balances for monitored bots and wallets.
    * @param tokenBalanceClient Instance of the TokenBalanceClient from the `financial-templates lib.
    * which provides synchronous access to address balances for a given expiring multi party contract.
-   * @param {List} botsToMonitor Array of bot objects to monitor. Each bot's `botName` `address`, `CollateralThreshold`
+   * @param {Object} config Object containing configuration for the balance monitor. Only option is a `botsToMonitor` 
+   * which is defines an array of bot objects to monitor. Each bot's `botName` `address`, `CollateralThreshold`
    *      and`syntheticThreshold` must be given. Example:
-   *      [{ name: "Liquidator Bot",
+   *      { botsToMonitor:[{ name: "Liquidator Bot",
    *         address: "0x12345"
    *         collateralThreshold: x1,
    *         syntheticThreshold: x2,
    *         etherThreshold: x3 },
-   *      ..]
+   *      ..] }
    * @param {Object} empProps Configuration object used to inform logs of key EMP information. Example:
    *      { collateralCurrencySymbol: "DAI",
             syntheticCurrencySymbol:"ETHBTC",
             priceIdentifier: "ETH/BTC",
             networkId:1 }
    */
-  constructor(logger, tokenBalanceClient, botsToMonitor, empProps) {
+  constructor(logger, tokenBalanceClient, config, empProps) {
     this.logger = logger;
 
     // Instance of the tokenBalanceClient to read account balances from last change update.
     this.client = tokenBalanceClient;
     this.web3 = this.client.web3;
 
-    // Bot addresses and thresholds to monitor.
-    this.botsToMonitor = botsToMonitor;
+    // Bot addresses and thresholds to monitor. If none provided then defaults to monitoring nothing. Configuration
+    // object must conform to correct structure with the right key valued pairs.
+    const defaultConfig = {
+      botsToMonitor: {
+        value: [],
+        isValid: x => {
+          // For the config to be valid it must be an array of objects with the right keys within the object as being
+          // the `name`, `address`, `collateralThreshold`, `syntheticThreshold` and `etherThreshold`.
+          return (
+            Array.isArray(x) && // the value of `botsToMonitor` must be an array of objects.
+            x.every(y => {
+              // Each object within the array must have the following keys.
+              return (
+                Object.keys(y).includes("name") &&
+                Object.keys(y).includes("address") &&
+                this.web3.utils.isAddress(y.address) && // `address` must be a valid Ethereum address.
+                Object.keys(y).includes("collateralThreshold") &&
```

Comment:
Since you're checking that the `address` is the correct type, is there any reason why you don't check that `syntheticThreshold` and `etherThreshold` are numbers?

Reply:
Fair enough. to be consistent we should check all types on all inputs. I have implemented this change to all fields.

Reply:
In this specific context I actually think the inputs should be `string`s, not numbers for `collateralThreshold` `syntheticThreshold` & `etherThreshold` ONLY because these are wei represention of numbers. If we wanted to input 50 Eth, let's say, putting in the number `50000000000000000000` is better represented as a string rather than a number as the implementation that uses it down stream in the bot will need to cast this to a bigNumber to actually work with it. The natural input for a bignumber is a `string`, not a `number`.  All other numerical inputs should be `number`s.

I tried encoding this as a number for example but then you have issues all over the place with needing to convert from `number`->`string`->`bigNumber`. JS does some nasty encoding with larger numbers like this:

```
Number("2000000000000000000000000").toString()
'2e+24'
```
which cant be converted into a big number.